domain/quote_simulator.go

```diff
@@ -0,0 +1,23 @@
+package domain
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/osmosis-labs/osmosis/osmomath"
+)
+
+// QuoteSimulator simulates a quote and returns the gas adjusted amount and the fee coin.
+type QuoteSimulator interface {
+	// SimulateQuote simulates a quote and returns the gas adjusted amount and the fee coin.
+	// CONTRACT:
+	// - Only direct (non-split) quotes are supported.
```

Comment:
Why only non-split routes are supported? Nit: typo in `Retursn`

Reply:
Splits require a separate message and increase the complexity drastically.

My ultimate goal is to integrate SQS into Polaris at the lowest cost, and then we can iterate on improvements, including adding splits